src/cirrus/lib/statedb.py

```diff
@@ -368,6 +368,35 @@ def set_invalid(self, payload_id: str, msg: str) -> str:
         logger.debug("set invalid", extra=key.update({'last_error': msg}))
         return response
 
+    def set_aborted(self, payload_id: str) -> str:
+        """Set this item as ABORTED
+
+        Args:
+            payload_id (str): The Cirrus Payload
+
+        Returns:
+            str: DynamoDB response
+        """
+        now = datetime.now(timezone.utc).isoformat()
+        key = self.payload_id_to_key(payload_id)
+
+        expr = (
+            'SET '
+            'created = if_not_exists(created, :created), '
+            'state_updated=:state_updated, updated=:updated'
+        )
+        response = self.table.update_item(
+            Key=key,
+            UpdateExpression=expr,
+            ExpressionAttributeValues={
+                ':created': now,
```

Comment:
So the query sets `created` if that `created` field doesn't already exist, and it used the value from `:created` if it needs to be set. Really, I'm not sure how we'd get to this `ABORTED` state without the record already existing and having the `created` field already, so in practice this query should never really set `created`, but it is safer to check in case we need to, I guess.

And I copied this from `set_invalid`, so if we want to change this for `ABORTED` I think we'd need to audit all the other state changes too.